src/shared/directives/substitutor/label-substitutor.directive.ts

```diff
@@ -59,7 +61,17 @@ export class LabelSubstitutorDirective implements OnInit, OnDestroy {
   }
 
   private substituteLabel(fields, label) {
-    return this.labelSubstitutionService.substituteLabel(fields, label);
+    return this.labelSubstitutionService.substituteLabel(fields, label, this.isEmptyIfPlaceholderMissing);
   }
 
+  private getLabel() {
+    return this.caseField.value || this.caseField.label;
+  }
+  private setLabel(label: string) {
```

Comment:
This method still bothers me. Forgetting for the moment that it actually works it seems to me we are trying to hack in a functionality using another one namely label substitution. It will be really hard for someone new to understand what happens here.

Reply:
Will remove this method and simplify.